src/dos/dos_keyboard_layout.cpp

```diff
@@ -226,7 +226,10 @@ static Bit32u read_kcl_file(const char* kcl_file_name, const char* layout_id, bo
 	return 0;
 }
 
-static Bit32u read_kcl_data(Bit8u * kcl_data, Bit32u kcl_data_size, const char* layout_id, bool first_id_only) {
+static Bit32u read_kcl_data(const std::vector<Bit8u> &kcl_data,
+                            const char *layout_id,
+                            bool first_id_only)
+{
```

Comment:
Prefer `uint32_t` and `uint8_t`

Is `first_id_only` used as a temporary (or written into) throughout the function? 
If it's held read-only, suggest `const`'ing it too at the API level.